src/components/RegistrationForm.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+import { DashboardProps } from "./Dashboard";
+import Router from "next/router";
+import { postEntry } from "@/utils/apiCalls";
+
+const RegistrationForm = ({isAuthorized, userID, data }: DashboardProps) => {
+  const [startDate, setStartDate] = useState<string | number | readonly string[] | undefined>(undefined);
+  const [periodLength, setPeriodLength] = useState<string | undefined>(undefined);
+ 
+  useEffect(() => {
+    if (!isAuthorized) {
+      Router.push("/");
```

Comment:
I think we should use `useRouter` here as well ( i mentioned it in another comment, I can easily fix this if this branch gets merged before I put up my PR)